### PR TITLE
Character, Friendly and Enemy Target classes

### PR DESCRIPTION
**Pull request by Mehuge**
Adds Character, FriendlyTarget and EnemyTarget Combatant based classes to represent these entities.

Alter clientInterface so that the OnCharacterRaceChanged callback takes a race: race parameter.

### CODE REVIEW

**Comment by Mehuge** [comment]
I wonder if FriendlyTarget and EnemyTarget would be better called just Friendly and Enemy?

File: src/ts/classes/Character.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import Combatant from './Combatant';
+import race from '../constants/race';
+import archetype from '../constants/archetype';
+
+class Character  extends Combatant {
```
Comment (saddieeiddas) [inline src/ts/classes/Character.ts]: double space after class name

**Comment by saddieeiddas** [comment]
I'm think `Friendly`/`Enemy` would be nice

Would this get confusing later on when there are lots of other classes, and could any of them classes need `Friendly` or `Enemy` in the name?

**Comment by codecorsair** [comment]
@saddieeiddas namespacing is your friend in this case.  

However, these classes, as is, have nothing to do with a specific "target ui" or anything.  It's just a representation of a character in game's data used for the UI.  Now, with this in mind, there is no purpose for the FriendlyTarget / EnemyTarget that I can see.  All the data needed for the UI lives in the Combatant class.

File: src/ts/classes/Character.ts (hunk 1 above)
Comment (codecorsair) [inline src/ts/classes/Character.ts]: I would also like this class to be called Player.  Makes more sense in line with the client side code definitions.

**Comment by codecorsair** [comment]
Go for it! :+1:

**Pull request merged**